### PR TITLE
Py23 more

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**

### CODE REVIEW

**Comment by jolyonb** [comment]
Excellent. I'll trade you a review of this for #214 ;-)

File: conftest.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import numpy as np
+
+try:
+    # Prior to version 1.13, numpy added an extra space before floats when printing arrays
+    # We use 1.6 for Python 2 and 1.16 for Python 3, so the printing difference
+    # causes problems for doctests.
+    #
+    # Setting the printer to legacy 1.13 combined with the doctest directive
+    # NORMALIZE_WHITESPACE is fixes the issue.
```
Comment (jolyonb) [inline conftest.py]: Extra `is` in this line.

**Comment by jolyonb** [comment]
This is getting to be a chunky PR - almost at 700 lines here. I figure there's more you want to do, but is this reviewable as is? (I'd prefer not to do a 1000 line PR...)

**Comment by ChristopherChudzicki** [comment]
Sure, this is ready for review. Most of the +500 –300 lines are reasonably trivial changes to the doctests.

**Comment by ChristopherChudzicki** [comment]
BTW: Only 8 Python 3 tests are failing. I'll tackle them tomorrow.

File: mitxgraders/helpers/get_number_of_args.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    # (in Python 2.7, unbound methods have __self__ = None)
+    try:
+        if func.__self__ is not None:
+            num_args += -1
+    except AttributeError:
+        pass
+
+    return num_args
+
+def get_number_of_args_py3(callable_obj):
```
Comment (ChristopherChudzicki) [inline mitxgraders/helpers/get_number_of_args.py]: @jolyonb BTW: This function and one line below are responsible for the drop in Python 2 coverage.

This function is probably not strictly necessary: `get_number_of_args_py2` works in Python 3 also, however it raises a `DeprecationWarning` for `getargspec`. 

Also, `get_number_of_args_py3` is about 100 times simpler, even with the annoying numpy special case.

File: mitxgraders/helpers/calc/math_array.py [hunk 3]
```diff
@@ -16,8 +16,8 @@ def is_number_zero(value):
     """
     Tests whether a value is the scalar number 0.
 
-    >>> map(is_number_zero, [0, 0.0, 0j])
-    [True, True, True]
+    >>> is_number_zero(0), is_number_zero(0.0), is_number_zero(0)
```
Comment (jolyonb) [inline mitxgraders/helpers/calc/math_array.py]: Missing a `j` at the end here.

**Comment by jolyonb** [comment]
Alrighty, will tackle this first thing tomorrow morning!

File: tests/formulagrader/test_formulagrader.py [hunk 4]
```diff
@@ -446,10 +446,18 @@ def is_coterminal_and_large(comparer_params, student_input, utils):
         reduced = student_input % (360)
         return utils.within_tolerance(answer, reduced) and student_input > min_value
 
-    mocked = mock.Mock(side_effect=is_coterminal_and_large,
-                # The next two kwargs ensure that the Mock behaves nicely for inspect.getargspec
-                spec=is_coterminal_and_large,
-                func_code=is_coterminal_and_large.func_code,)
+    mocked = (mock.create_autospec(is_coterminal_and_large,
+                                   side_effect=is_coterminal_and_large)
+              if six.PY3 else
```
Comment (jolyonb) [inline tests/formulagrader/test_formulagrader.py]: Yes, this works, but I'd much prefer a normal if statement here for readability.

File: mitxgraders/helpers/get_number_of_args.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    NOTES:
+        - Seems to work in Python 3, but is based on inspect.getargspec which
+          raises a DeprecationWarning in Python 3.
+        - in Python 3, use the much simpler signature-based
+          get_number_of_args_py3 funciton instead.
+        - Cannot handle class constructors
+
+    Usage
+    =====
+    See documetnation for get_number_of_args.
```
Comment (jolyonb) [inline mitxgraders/helpers/get_number_of_args.py]: Nit: `documentation`

File: mitxgraders/helpers/get_number_of_args.py [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+    return num_args
+
+def get_number_of_args_py3(callable_obj):
+    """
+    Get number of positional arguments of function or callable object.
+
+    NOTES:
+        - based on inspect.signature
+        - in Python 2, use getargspec-based get_number_of_args_py3 instead
```
Comment (jolyonb) [inline mitxgraders/helpers/get_number_of_args.py]: Should be py2 at the end?

**Pull request merged**